engine/src/sysunxdate.cpp

```diff
@@ -230,8 +258,16 @@ static void cache_locale(void)
     }
     else
     {
-        s_datetime_locale -> time_formats[0] = string_prepend(swap_time_tokens(t_time_ampm), '!');
+        // PM-2015-09-07: [[ Bug 9942 ]] On Linux, the short/abbr system time
+        //  should not return the seconds
+        MCStringRef t_short_time;
+        t_short_time = string_prepend(swap_time_tokens(t_time_ampm), '!');
+
+        s_datetime_locale -> time_formats[0] = remove_seconds(t_short_time);
         s_datetime_locale -> time_formats[1] = swap_time_tokens(t_time_ampm);
+
+        // string_prepend() returns a new StringRef, which we must release
+        MCValueRelease(t_short_time);
```

Comment:
Yes: `string_prepend` returns a new instance, and `MCAutoStringRef::operator =` retains the right operand.
The best would certainly to change `string_prepend` to return the string through a return parameter, in which case we can use `MCAutoStringRef::operator &` - which does not retain.